DAP Version compatibility (#568)
ietf-wg-ppm/draft-ietf-ppm-dap: draft-ietf-ppm-dap.md

# IANA Considerations

> ## Message versioning

Reviewer: ```suggestion
## Message versioning

```

> Media types for the HTTP requests are specific to this version of DAP. When a new major enhancement is proposed that results in newer IETF specification for DAP, a new set of media types need to be defined. In other words, newer versions of DAP will not be backward compatible with this version of DAP.

Reviewer: Wrap paragraphs at 80 columns.

Author: done

> HTTP requests with DAP media types MAY express an optional parameter 'version'. Value of this parameter indicates current draft version of the protocol the component is using. This MAY be used as a hint by the receiver of the request to do compatibility checks between client and server.

Reviewer 2: I think this paragraph would not fit into the eventual RFC for DAP, since there will be no draft version to refer to. IMO, add a TODO to remove this before RFC.

Reviewer: Agreed. Concretely, add something like this: 
```suggestion
(NOTE TO RFC EDITOR: Remove this paragraph.) HTTP requests with DAP media types MAY express an optional parameter 'version'. Value of this parameter indicates current draft version of the protocol the component is using. This MAY be used as a hint by the receiver of the request to do compatibility checks between client and server.
```

Reviewer 3: I'd like to see an example of this optional `version` parameter. Would this be something like `Media-Type: application/dap-aggregation-job-init-req;version=11`? A reference to [RFC 9110 section 8.3](https://www.rfc-editor.org/rfc/rfc9110.html#section-8.3.1) (which describes this kind of thing in media types) also would help.

Author: incorporated both suggestions.

# IANA Considerations

> ## Message versioning

Reviewer: Let's move this to a subsection of `## Protocol Message Media Types` immediately below this.

Author: Moved.

Media types for the HTTP requests are specific to this version of DAP. When a 
new major enhancement is proposed that results in newer IETF specification for 
DAP, a new set of media types need to be defined. In other words, newer 
versions of DAP will not be backward compatible with this version of DAP.

(NOTE TO RFC EDITOR: Remove this paragraph.) HTTP requests with DAP media types
MAY express an optional parameter 'version', following {{Section 8.3 of !RFC9110}}.
Value of this parameter indicates current draft version of the protocol the 
component is using. This MAY be used as a hint by the receiver of the request 
to do compatibility checks between client and server.

> For example, A report submission to leader from a client that supports DAP09
> would have the header Media-Type: application/dap-report;version=09.

Reviewer 3: ```suggestion
For example, a report submission to leader from a client that supports
draft-ietf-ppm-dap-09 could have the header `Media-Type:
application/dap-report;version=09`.
```

Prefer notation `draft-ietf-ppm-dap-09` because that's a bit more formal in IETF parlance. Also use "could" instead of "would", because this text isn't normative: it's completely up to implementations what to put in the `version` field and how to interpret it.

Author: Sounds good. Done.

[OPEN ISSUE: Solicit review of these allocations from domain experts.]

### Message versioning

Media types for the HTTP requests are specific to this version of DAP. When a 
new major enhancement is proposed that results in newer IETF specification for 
DAP, a new set of media types need to be defined. In other words, newer 
versions of DAP will not be backward compatible with this version of DAP.

(NOTE TO RFC EDITOR: Remove this paragraph.) HTTP requests with DAP media types
MAY express an optional parameter 'version', following {{Section 8.3 of !RFC9110}}.

> Value of this parameter indicates current draft version of the protocol the 

Reviewer: ```suggestion
Value of this parameter indicate current draft version of the protocol the 
```

> component is using. This MAY be used as a hint by the receiver of the request

Reviewer: This is non-normative IMO
```suggestion
component is using. This might be used as a hint by the receiver of the request
```
